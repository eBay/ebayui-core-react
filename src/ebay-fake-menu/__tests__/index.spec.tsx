import React  from 'react'
import { fireEvent, render } from '@testing-library/react';
import initStoryshots from '@storybook/addon-storyshots'
import { EbayFakeMenu, EbayFakeMenuItem } from '../index'

initStoryshots({
    config: ({ configure }) => configure(() => require('./index.stories'), module)
})

describe('<EbayFakeMenu>', () => {
    describe('on menu item click', () => {
        it('should fire onClick event', () => {
            const onClickSpy = jest.fn()
            const wrapper = render(
                <EbayFakeMenu>
                    <EbayFakeMenuItem href="#" onClick={onClickSpy}/>
                </EbayFakeMenu>
            )

            fireEvent.click(wrapper.container.querySelectorAll('.fake-menu__item')[0])

            expect(onClickSpy).toBeCalled()
        })
    })
    describe('on key down', () => {
        it('should fire onKeyDown event', () => {
            const onKeyDownSpy = jest.fn()
            const wrapper = render(
                <EbayFakeMenu onKeyDown={onKeyDownSpy}>
                    <EbayFakeMenuItem href="#" />
                </EbayFakeMenu>
            )

            fireEvent.keyDown(wrapper.container.querySelectorAll('.fake-menu__item')[0])

            expect(onKeyDownSpy.mock.calls[0][1]).toEqual(0)
        })
    })
    describe('on item select', () => {
        it('should fire onSelect event', () => {
            const onSelectSpy = jest.fn()
            const wrapper = render(
                <EbayFakeMenu onSelect={onSelectSpy}>
                    <EbayFakeMenuItem href="#" />
                    <EbayFakeMenuItem href="#" />
                </EbayFakeMenu>
            )

            fireEvent.click(wrapper.container.querySelectorAll('.fake-menu__item')[1])

            expect(onSelectSpy).toBeCalledWith(expect.anything(), 1)
        })
    })

    describe('on appear', () => {
        it('should not autofocus', () => {
            const wrapper = render(
                <EbayFakeMenu>
                    <EbayFakeMenuItem href="#" />
                    <EbayFakeMenuItem href="#" />
                </EbayFakeMenu>
            )
            const firstMenuItem = wrapper.container.querySelectorAll('.fake-menu__item')[0]

            expect(firstMenuItem).not.toHaveFocus()
        })
        it('should autofocus on first item', () => {
            const wrapper = render(
                <EbayFakeMenu>
                    <EbayFakeMenuItem href="#" autoFocus />
                    <EbayFakeMenuItem href="#" />
                </EbayFakeMenu>
            )
            const firstMenuItem = wrapper.container.querySelectorAll('.fake-menu__item')[0]

            expect(firstMenuItem).toHaveFocus()
        })
    })
})
