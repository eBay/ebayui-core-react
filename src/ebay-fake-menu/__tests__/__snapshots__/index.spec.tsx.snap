// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-fake-menu Default 1`] = `
<div
  className="fake-menu"
  onClick={[Function]}
>
  <ul
    className="fake-menu__items"
    tabIndex={-1}
  >
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Current page
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Storyshots ebay-fake-menu With Disabled Item 1`] = `
<div
  className="fake-menu"
>
  <ul
    className="fake-menu__items"
    tabIndex={-1}
  >
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-disabled="true"
        className="fake-menu__item"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Storyshots ebay-fake-menu With Separator 1`] = `
<div
  className="fake-menu"
>
  <ul
    className="fake-menu__items"
    tabIndex={-1}
  >
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <hr
        className="fake-menu__separator"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="separator"
      />
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 4
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 5
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Storyshots ebay-fake-menu Without tick icon 1`] = `
<div
  className="fake-menu"
  onClick={[Function]}
>
  <ul
    className="fake-menu__items"
    tabIndex={-1}
  >
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-current="true"
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Current page
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        className="fake-menu__item"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span>
          Item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;
